packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/nlb/network-load-balancer.ts

```diff
@@ -288,6 +302,16 @@ export class NetworkLoadBalancer extends BaseLoadBalancer implements INetworkLoa
    * @returns The newly created listener
    */
   public addListener(id: string, props: BaseNetworkListenerProps): NetworkListener {
+    // UDP listener with dual stack NLB requires prefix IPv6 source NAT to be enabled
+    if (
+      (props.protocol === Protocol.UDP || props.protocol === Protocol.TCP_UDP) &&
+      (this.ipAddressType === IpAddressType.DUAL_STACK || this.ipAddressType === IpAddressType.DUAL_STACK_WITHOUT_PUBLIC_IPV4)
+    ) {
+      if (this.enablePrefixIpv6SourceNat === false) {
+        throw new Error('To add a listener with UDP protocol to a dual stack NLB, \'enablePrefixForIpv6SourceNat\' must be set to true.');
+      }
+      this.enablePrefixIpv6SourceNat = true;
```

Comment:
When creating a UDP listener for a dual-stack NLB, enabling `enablePrefixIpv6SourceNat ` is mandatory, so I modified the `addListener()` function to automatically configure it.

Reply:
I don't like implicit behaviour. Since we already have an error case right above, IMO we should check if `enablePrefixIpv6SourceNat` is not defined or `false`, then throw an error. The error msg should be clear enough to let users understand to set `enablePrefixIpv6SourceNat` to true

Reply:
I agree with you! I've updated my code.